osu.Game/Screens/Play/HUD/ClicksPerSecond/ClicksPerSecondCounter.cs

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Sprites;
+using osu.Framework.Localisation;
+using osu.Game.Graphics;
+using osu.Game.Graphics.Sprites;
+using osu.Game.Graphics.UserInterface;
+using osu.Game.Skinning;
+using osuTK;
+
+namespace osu.Game.Screens.Play.HUD.ClicksPerSecond
+{
+    public class ClicksPerSecondCounter : RollingCounter<int>, ISkinnableDrawable
+    {
+        private const float alpha_when_invalid = 0.3f;
+
+        private readonly Bindable<bool> valid = new Bindable<bool>();
+
+        private ClicksPerSecondCalculator? calculator;
+
+        protected override double RollingDuration => 350;
+
+        public bool UsesFixedAnchor { get; set; }
+
+        public ClicksPerSecondCounter()
+        {
+            Current.Value = 0;
+        }
+
+        [BackgroundDependencyLoader]
+        private void load(OsuColour colours, ClicksPerSecondCalculator calculator)
+        {
+            this.calculator = calculator;
+            Colour = colours.BlueLighter;
+            valid.BindValueChanged(e =>
+                DrawableCount.FadeTo(e.NewValue ? 1 : alpha_when_invalid, 1000, Easing.OutQuint));
+        }
+
+        protected override void Update()
+        {
+            base.Update();
+
+            valid.Value = calculator != null && calculator.Ready;
+            Current.Value = calculator != null ? calculator.Ready ? calculator.Value : 0 : 0;
+        }
+
+        protected override IHasText CreateText() => new TextComponent
+        {
+            Alpha = alpha_when_invalid
+        };
+
+        private class TextComponent : CompositeDrawable, IHasText
+        {
+            public LocalisableString Text
+            {
+                get => text.Text;
+                set => text.Text = value;
+            }
+
+            private readonly OsuSpriteText text;
+
+            public TextComponent()
+            {
+                AutoSizeAxes = Axes.Both;
+
+                InternalChild = new FillFlowContainer
+                {
+                    AutoSizeAxes = Axes.Both,
+                    Spacing = new Vector2(2),
+                    Children = new Drawable[]
+                    {
+                        text = new OsuSpriteText
+                        {
+                            Anchor = Anchor.BottomLeft,
+                            Origin = Anchor.BottomLeft,
```

Comment:
This doesn't work great if you make the counter right aligned (it may expand off the edge of the screen). These should probably be made aware of the user's chosen alignment and expand in the correct direction (or enough padding should be added so it doesn't need to expand).

Reply:
Yeah, I just copied what was in `PerformancePointsCounter` and `UnstableRateCounter`, I did not think about that. I will look into this.

Reply:
If it also faces the same issue, it might be fine to leave it for now and fix them together in a follow-up, in that case.